src/every.ts

```diff
@@ -0,0 +1,56 @@
+import { purry } from "./purry";
+
+/**
+ * Checks if every element of `data` passed the predicate `predicate`.
+ *
+ * @param data - The Array to check.
+ * @param predicate - A predicate function to run.
+ * @returns A `boolean` indicating if every element in `data` passes the predicate `pred`.
+ * @signature
+ *    R.every(arr, predicate)
+ * @example
+ *    R.every([1, 2, 3], R.isNumber) // => true
+ *    R.every(['a', 0, 'b'], R.isString) // => false
+ * @dataFirst
+ * @indexed
+ * @category Array
+ */
+export function every<T, S extends T>(
+  data: ReadonlyArray<T>,
+  predicate: (input: T, index: number, array: ReadonlyArray<T>) => input is S,
+): data is Array<S>;
```

Comment:
The typing does not respect the shape of the input in the output. We can take the tuple as a generic parameter and use it to return a result that is the same shape.

```suggestion
export function every<T extends IterableContainer, S extends T[number]>(
  data: T,
  predicate: (input: T[number], index: number, array: T) => input is S,
): data is { [K in keyof T]: S };
```

This would allow someone to test something like `[string | number, ...Array<string | number>]` and get back `[string, ...Array<string>]`

The non-narrowing overload doesn't need this treatment because it returns a `boolean`